src/Wave/DB/Query.php

```diff
@@ -214,10 +215,7 @@ public function join($type, $class, &$alias = null, &$target_alias = null) {
 
         $this->resolveNamespace($class);
 
-        //@patrick, should this not actually select these rows? Should it be an extra parameter in the function constructor?
-        //$join_table_alias = $this->addFieldsToSelect($class) could be replaced with:
-        //$join_table_alias = $this->aliasClass($class); // so it didn't add the fields but still aliased the table
-        if(!$this->manual_select)
+        if(!$this->manual_select && $hydrate === true)
```

Comment:
@calcinai turns out I did think about not selecting join data. I think this is doing the job unless I'm mistaken?

Reply:
So it does! Have you profiled a query to see how it goes?  The only other thing to consider, is the additional rows that are pulled when there's a join and ensure the logic to skip duplicates works

Reply:
Discussed this with @phindmarsh today as well. I think there are roughly three solutions. 

1 – Simplest option (possibly backwards incompatible, and a bit inefficient) – keep the loops inside `fetchRow` that assemble child rows / joins as is, and deal with duplicates by fingerprinting each object as it's being added to `joined_objects` in the model, and skip any that are already present (with a message indicating this has occurred in dev).  

We may need this regardless to deal with the situation where there are multiple joins and all but the last end up added to joined_objects more than once. 

2 – Add functionality into fetchRow that calculates a fingerprint comprised of the identifying columns of all of the child entities in a row, and keeps hold of this per result to ensure we don't add duplicates where we're not selecting from some of the joins. 

3 – Go with something more similar to the original design, and make a way to disable join selection+hydration (where the join is not a with()) for an entire query? Perhaps this is too simplistic and there is legitimacy in turning it on or off per join.

Reply:
I think I'd prefer #1, and using the existing `_getIdentifyingColumns()` method to decide if the row should be skipped or not.  If it's done after the joins are processed, it should be pretty simple, as the cursor will have already advanced.

Reply:
I've implemented the above duplicate detection – keen for feedback when you guys are free to review